pkgs/development/tools/phpactor/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib, stdenvNoCC, fetchFromGitHub, php, composer }:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "phpactor";
+  version = "2023.01.21";
+
+  src = fetchFromGitHub {
+    owner = "phpactor";
+    repo = "phpactor";
+    rev = version;
+    hash = "sha256-jWZgBEaffjQ5wCStSEe+eIi7BJt6XAQFEjmq5wvW5V8=";
+  };
+
+  buildInputs = [
+    composer
+    php
+  ];
+
+  buildPhase = ''
+    patchShebangs .
+    substituteInPlace composer.json \
+      --replace '"config": {' '"config": { "autoloader-suffix": "Phpactor",' \
+      --replace '"name": "phpactor/phpactor",' '"name": "phpactor/phpactor", "version": "${version}",'
+
+    composer dump-autoload --optimize --no-dev
```

Comment:
the `dump-autoload` step is not required here.

I succeeded to reproduce builds using the fixed-output hash: `sha256-iIy589rVG384qBr5bHdnXvzhkIXuYldEx6LLvGNDN5w=`

Reply:
I get the same output hash. Either Ryan forgot to update it, or it is still not reproducible enough.

Reply:
Oh, right the derivation includes paths to other derivational in shebangs so FOD cannot really be used for the package. At best, you could create a FOD for the vendor directory.